Hardcoded wasm-opt path invalid when installing jco as a dev dep

I've installed jco as a dev time dependency instead of global bin (`npm i -D @bytecodealliance/jco`).

The `opt` command looks for `wasm-opt` at `../../node_modules/binaryen/bin/wasm-opt`.

However, npm installed it at the root of three to avoid duplicates (`node_modules/binaryen`, not `node_modules/@bytecodealliance/jco/node_modules/binaryen`).

Perhaps, using `require.resolve('binaryen/bin/wasm-opt')` will resolve the issue.

Full error text:

```
(jco opt) Error: node:internal/modules/cjs/loader:1152
  throw err;
  ^

Error: Cannot find module '***/example/node_modules/@bytecodealliance/jco/node_modules/binaryen/bin/wasm-opt'
    at Module._resolveFilename (node:internal/modules/cjs/loader:1149:15)
    at Module._load (node:internal/modules/cjs/loader:990:27)
    at Function.executeUserEntryPoint [as runMain] (node:internal/modules/run_main:142:12)
    at node:internal/main/run_main_module:28:49 {
  code: 'MODULE_NOT_FOUND',
  requireStack: []
}

Node.js v21.6.1
```

--- a/packages/jco-transpile/src/opt.js
+++ b/packages/jco-transpile/src/opt.js
@@ -11,6 +11,7 @@ const { metadataShow, print } = tools;
  *  asyncify?: boolean,
  *  optArgs?: string[],
  *  noVerify?: boolean
+ *  wasmOptBin?: string
  * }} OptimizeOptions
  */
 
@@ -65,7 +66,8 @@ export async function runOptimizeComponent(componentBytes, opts) {
                         metadata.range[0],
                         metadata.range[1]
                     ),
-                    args
+                    args,
+                    opts
                 );
 
                 // compute the size change, including the change to
@@ -188,15 +190,16 @@ export async function runOptimizeComponent(componentBytes, opts) {
  *
  * @param {Uint8Array} source
  * @param {Array<string>} args
+ * @param {TranspileOpt} transpileOpts
  * @returns {Promise<Uint8Array>}
  */
-async function runWasmOptCLI(source, args) {
-    const wasmOptPath = fileURLToPath(
-        import.meta.resolve('binaryen/bin/wasm-opt')
-    );
+async function runWasmOptCLI(source, args, transpileOpts) {
+    const wasmOptBin =
+        transpileOpts?.wasmOptBin ??
+        fileURLToPath(import.meta.resolve('binaryen/bin/wasm-opt'));
 
     try {
-        return await runWASMTransformProgram(wasmOptPath, source, [
+        return await runWASMTransformProgram(wasmOptBin, source, [
             ...args,
             '-o',
         ]);

--- a/packages/jco/src/cmd/opt.js
+++ b/packages/jco/src/cmd/opt.js
@@ -136,7 +136,8 @@ export async function optimizeComponent(componentBytes, opts) {
                             metadata.range[0],
                             metadata.range[1]
                         ),
-                        args
+                        args,
+                        opts
                     );
 
                     // compute the size change, including the change to
@@ -280,13 +281,13 @@ export async function optimizeComponent(componentBytes, opts) {
  * @param {Array<string>} args
  * @returns {Promise<Uint8Array>}
  */
-async function wasmOpt(source, args) {
-    const wasmOptPath = fileURLToPath(
-        import.meta.resolve('binaryen/bin/wasm-opt')
-    );
+async function wasmOpt(source, args, transpileOpts) {
+    const wasmOptBin =
+        transpileOpts?.wasmOptBin ??
+        fileURLToPath(import.meta.resolve('binaryen/bin/wasm-opt'));
 
     try {
-        return await spawnIOTmp(wasmOptPath, source, [...args, '-o']);
+        return await spawnIOTmp(wasmOptBin, source, [...args, '-o']);
     } catch (e) {
         if (e.toString().includes('BasicBlock requested')) {
             return wasmOpt(source, args);

--- a/packages/jco/src/cmd/transpile.js
+++ b/packages/jco/src/cmd/transpile.js
@@ -122,6 +122,7 @@ async function wasm2Js(source) {
  *   multiMemory?: bool,
  *   experimentalIdlImports?: bool,
  *   optArgs?: string[],
+ *   wasmOptBin?: string[],
  * }} opts
  * @returns {Promise<{ files: { [filename: string]: Uint8Array }, imports: string[], exports: [string, 'function' | 'instance'][] }>}
  */

--- a/packages/jco/src/jco.js
+++ b/packages/jco/src/jco.js
@@ -234,6 +234,7 @@ program
         []
     )
     .option('--all-features', 'enable all features')
+    .option("--wasm-opt-bin <path-to-wasm-opt>', 'wasm-opt binary path (default: '')")
     .action(asyncAction(types));
 
 program
@@ -345,6 +346,7 @@ program
     )
     .option('--asyncify', 'runs Asyncify pass in wasm-opt')
     .option('-q, --quiet')
+    .option("--wasm-opt-bin <path-to-wasm-opt>', 'wasm-opt binary path (default: '')")
     .allowExcessArguments(true)
     .action(asyncAction(opt));
 

